Sources/AppcuesKit/Presentation/ExperienceRendering/ExperienceStateMachine+State.swift

```diff
@@ -154,7 +154,8 @@ extension ExperienceStateMachine.Transition {
             sideEffect = .dismissContainer(package, continuation: .startStep(stepRef))
         }
 
-        return .init(toState: .endingStep(experience, stepIndex, package), sideEffect: sideEffect)
+        // Moving to a new step is an interaction that indicates the ending step is completed
+        return .init(toState: .endingStep(experience, stepIndex, package, markComplete: true), sideEffect: sideEffect)
```

Comment:
@mmaatttt noting something I found while testing related things on the Android side - if I have a button that goes backwards, previous step - it will trigger step completion, since `markComplete` is always true during this transition 🤔

Reply:
update: discussion and updated reqs captured here https://app.shortcut.com/appcues/story/41699/do-not-trigger-step-completed-when-flow-is-dismissed#activity-42263 (and in linked ios ticket)

Reply:
Right, makes sense. I guess combined with https://app.shortcut.com/appcues/story/41902/add-step-interaction-tracking-for-mobile-button-interactions, it should be a `step_interaction` but not `step_completed`?

Reply:
yeah, that makes sense.